specs/Deployment/DeploymentAPI.md

```diff
@@ -0,0 +1,141 @@
+scenario will have permissions to deploy them, we will minimize developer and user pain by
+including the main package MSIX packages within the framework, and this API will allow for the
+deployment of that package if needed.
+
+Processes lacking permission to deploy MSIX are out-of-scope at this time.
+
+# Examples
+
+## GetStatus
+
+This is the means by which an app or program can call to verify all required packages are present
+to a minimum version needed by the loaded framework.  A process should call GetStatus() after
+process initialization and before using Windows App Runtime content (e.g. in main() or WinMain()).
+
+Developers call this method without parameters, as all information about the version, channel, and
+architecture are derived from the current WARfwk loaded. This method is good to identify if there
+is possible work required in the form of package install which may take several seconds to
+complete.
+
+```C++
+auto status{Microsoft::Windows::ApplicationModel::WindowsAppRuntime::Deployment::GetStatus()};
+if (!status.IsOK())
+{
+    ...
+}
+```
+
+## Initialize
+
+This API performs a status check and if any packages are required for install, attempts to register
+those packages and get the WindowsAppRuntime into an OK state. This will perform package install if
+possible. It is possible multiple packages may be required for install, and this will attempt
+to deploy all required packages. The returned WindowsAppRuntimeStatus will contain the first error
+if one occurs during package install. There will be no rollback of successfully installed packages
+if an error occurs on a subsequent package install.
+
+
+```C++
+auto status{Microsoft::Windows::ApplicationModel::WindowsAppRuntime::Deployment::Initialize()};
+if (!status.IsOK())
+{
+    return status.ExtendedError();
+}
+```
+
+
+# API Details
+
+```c#
+namespace Microsoft.Windows.ApplicationModel.WindowsAppSDK
+{
+    /// Represents the current Deployment status of the WindowsAppRuntime
+    [experimental]
+    runtimeclass DeploymentStatus
+    {
+        DeploymentStatus(
+            Boolean isOK,
+            Boolean packageInstallRequired,
+            HRESULT extendedError);
+
+        /// Returns True if WindowsAppRuntime has all required packages at equal or greater version
+        /// than the current framework requires.
+        Boolean IsOK{ get; };
+
+        /// Returns True if a Package install is required the WindowsAppRuntime to be in a good
+        /// state.
+        Boolean PackageInstallRequired{ get; };
+ 
+        /// Returns the first encountered error if there was an error initializing or getting the status,
+        /// or S_OK if no error.
+        HRESULT ExtendedError{ get; };
+    }
+
+    /// Used to query deployment information for WindowsAppRuntime
+    [experimental]
+    static runtimeclass DeploymentManager
+    {
+        /// Returns the current deployment status of Windows App Runtime package full name provided.
```

Comment:
Does this API work for any package full name? Or just the ones that are part of WinAppSDK?

What if the package referenced is superseded by another package that was deployed by servicing?